src/stories/auth/basicAuth.ts

```diff
@@ -0,0 +1,32 @@
+const authProvider = {
+  login: ({ username, password }: { username: string; password: string }) => {
+    if (username !== 'john' || password !== '123') {
+      return Promise.reject();
+    }
+    localStorage.setItem('username', username);
+    return Promise.resolve();
+  },
+  logout: () => {
+    localStorage.removeItem('username');
+    return Promise.resolve();
+  },
+  checkAuth: () =>
+    localStorage.getItem('username') ? Promise.resolve() : Promise.reject(),
+  checkError: (error: { status: number }) => {
+    const { status } = error;
+    if (status === 401 || status === 403) {
+      localStorage.removeItem('username');
```

Comment:
nitpick: not sure this line is necessary. React Admin will already call `logout` if `checkError` rejects.